### PR TITLE
refactor: use PDM

### PR DESCRIPTION
**Pull request by elenakrittik**
This PR sets up the repo to use [PDM](https://pdm.fming.dev/), similarly to disnake.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -133,4 +133,7 @@ types
 test.py
 
 # IDE config
-.vscode
\ No newline at end of file
+.vscode
+
+# PDM
+.pdm.toml
```
Comment (Snipy7374) [inline .gitignore:139]: We need to ignore `pdm.lock` too

**Review by Snipy7374 — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (elenakrittik) [reply .gitignore:139]: No, we do not. pdm.lock is necessary to keep dependencies at their exact versions, just like poetry's lock files. See [docs](https://pdm.fming.dev/latest/usage/project/#working-with-version-control) on that.

Comment (elenakrittik) [reply .gitignore:139]: It was decided to gitignore pdm.lock *for now*.

**Review by Snipy7374 — APPROVED** [review]
Thanks!

**Pull request merged**